src/core/contracts/identity.ts

```diff
@@ -183,6 +192,117 @@ const getBeaconAddress = async (opts?: ConfigOpts): Promise<EthereumAddress> =>
   return address;
 };
 
+/**
+ * DelegateAdd represents a struct for adding delegates
+ */
+interface DelegateAdd extends TypedMessageDataField {
+  nonce: number;
+  delegateAddr: HexString;
+  role: DelegationRole;
+}
+
+/**
+ * DelegateAddParams represents a partial struct for adding delegates
+ */
+export interface DelegateAddParams {
+  nonce?: number;
+  delegateAddr: HexString;
+  role: DelegationRole;
+}
+
+/**
+ * getDomainSeparator() Gets unique data that identifies the Identity contract to prevent phishing attacks
+ *
+ * @param contractAddress - Address of the identity contract to use
+ * @param opts - Optional. Configuration overrides, such as store, if any
+ * @returns A EIP-712 domain structure that is unique to the Identity contract
+ */
+export const getDomainSeparator = async (
+  contractAddress: EthereumAddress,
+  opts?: ConfigOpts
+): Promise<TypedDomainData> => {
+  const provider = requireGetProvider(opts);
+
+  return {
+    name: "Identity",
+    version: "1",
+    chainId: (await provider.getNetwork()).chainId,
+    verifyingContract: contractAddress,
+    salt: "0xa0bec69846cdcc8c1ba1eb93be1c5728385a9e26062a73e238b1beda189ac4c9",
+  };
+};
+
+/**
+ * createAddDelegateEip712TypedData() Allows users to create typed structure data for signing addDelegate messages
+ * following the EIP-712 standard.
+ *
+ * @param contractAddress - Address of the identity contract to use
+ * @param message - The DelegateAdd message to be signed
+ * @param opts - Optional. Configuration overrides, such as store, if any
+ * @returns Typed structure data for signing addDelegate messages
+ */
+export const createAddDelegateEip712TypedData = async (
```

Comment:
Non-blocking: I wonder if we might want to move this and `getDomainSeparator` to the utilities as well?

Reply:
Thats a good idea.